### ISSUE TITLE
0 new album or artist in index.php#stats.php?action=newest

### ISSUE THREAD

**Issue by douglasriper** [ampache/ampache#1977]
# Issue Template

## Environment

* Ampache version: 4.0 Dev (up to date)
* Web server + version: Apache + Mysql 8
* Server operating system: Ubunutu 18.04
* Client operating system: Manjaro/Firefox 58 , Android Dsub

* Client type:
  * Web
  * SubSonic
 
## Description
In the "Artist" and "Album" sections of the "New" page there are 0 items.
I looked in the database there is indeed the date indicated and the fields seem to me correctly filled in.
In addition, the new section in dSub is empty.

## Logs
No error in logs when I dispplay the page.

**Comment by lachlan-00** [comment]
That page should be presenting all albums/artists and sorting by newest first with no limit

Does the mashup show anything for newest?
index.php#mashup.php?action=album
index.php#mashup.php?action=artist

**Comment by douglasriper** [comment]
The pages display 0 for new artists and new albums. 
But in these two pages the other sections work.

I searched for the sql query executed but I'm not good at php (＃＞＜)

**Comment by lachlan-00** [comment]
I can tell you what the queries are. Depending on your settings there are a few different queries that run so we should check them all.

get_newest_sql ALBUM (rating_filter set, Group Disks)
```
SELECT DISTINCT(`album`) as `id`, `addition_time` AS `real_atime`
FROM `song`
LEFT JOIN `album` ON `album`.`id` = `song`.`album`
WHERE `song`.`album` NOT IN (SELECT `object_id` FROM `rating`
                             WHERE `rating`.`object_type` = 'album' AND `rating`.`rating` <=1 AND `rating`.`user` = 2)
GROUP BY `album`.`name`, `album`.`album_artist`, `album`.`mbid`
ORDER BY `real_atime` DESC  
```

get_newest_sql ALBUM (No filter or Grouping)
```
SELECT DISTINCT(`album`) as `id`, `addition_time` AS `real_atime`
FROM `song`
ORDER BY `real_atime` DESC  
```

get_newest_sql ARTIST (rating_filter set)
```
SELECT DISTINCT(`artist`) as `id`, `addition_time` AS `real_atime`
FROM `song`
WHERE `song`.`artist` NOT IN (SELECT `object_id` FROM `rating`
                              WHERE `rating`.`object_type` = 'artist' AND `rating`.`rating` <=1 AND `rating`.`user` = 2)
GROUP BY `song`.`artist` ORDER BY `real_atime` DESC  
```

get_newest_sql ARTIST (No Filter)
```
SELECT DISTINCT(`artist`) as `id`, `addition_time` AS `real_atime`
FROM `song`
ORDER BY `real_atime` DESC  
```

**Comment by douglasriper** [comment]
Thank you, I think I found the problem.
> SQL Error [1055] [42000]: Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'ampache.song.addition_time' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

Si j'ajoute le champs dans le GROUP la requete retourne les données.
Je vais essayé de désactiver only_full_group_by

**Comment by douglasriper** [comment]
By default Mysql seems to have only_full_group_by, by modifying mysqld.cnf (add the line sql_mode = "list of existing modes without only_full_group_by" I bypassed the problem.

Maybe we should modify the group closures if Mysql gets strict on that?

**Comment by lachlan-00** [comment]
yes we should!

So if seems that it's like TSQL/MSSQL where you need to either group or action the columns.

I think if we just set  MIN/MAX addition time then it will work for you.
```
SELECT DISTINCT(`album`) as `id`, MIN(`addition_time`) AS `real_atime`
FROM `song`
LEFT JOIN `album` ON `album`.`id` = `song`.`album`
GROUP BY `album`.`name`, `album`.`album_artist`, `album`.`mbid`
ORDER BY `real_atime` DESC 
```

**Comment by douglasriper** [comment]
it seems that the "select" must be in the "group" when only_full_group_by is activated.
> SQL Error [1055] [42000]: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'ampache.song.album' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

if I add to your request `song`. `album` in GROUP I have the list

**Comment by lachlan-00** [comment]
Can you add this to stats.class on your site and test?
![image](https://user-images.githubusercontent.com/1305249/62621089-64c1d600-b95e-11e9-9e5d-9bf44a870100.png)
```
$sql .= "GROUP BY `$sql_type`, `album`.`name`, `album`.`album_artist`, `album`.`mbid` ORDER BY `real_atime` DESC ";
```

**Comment by douglasriper** [comment]
Hi,
Of course, of course,
Tonight or tomorrow, I also have to think about changing the mysql configuration.
It's to remind me to do it 😉

**Comment by lachlan-00** [comment]
Thank you, i'm just testing a few changes in other areas before i run this on my servers.

**Comment by douglasriper** [comment]
I tried that in Dbbeaver and it works.
	SELECT DISTINCT(`album`) as `id`, MIN(`addition_time`) AS `real_atime` FROM `song` LEFT JOIN `album` ON `album`.`id` = `song`.`album` GROUP BY `album`, `album`.`name`, `album`.`album_artist`, `album`.`mbid` ORDER BY `real_atime` DESC

But by putting these lines in stats.class.php I get the message from the beginning
 if (AmpConfig::get('album_group') && $type == 'album') { $sql .= "GROUP BY `$sql_type`, `album`.`name`, `album`.`album_artist`, `album`.`mbid` ORDER BY `real_atime` DESC "; } else { $sql .= "GROUP BY `$sql_type` ORDER BY `real_atime` DESC ";

**Comment by lachlan-00** [comment]
Thanks for checking. Everything I have is mariadb so it takes a bit to test out MySQL.

I'll double check for other similar queries and put it in

**Comment by douglasriper** [comment]
If I can help, no problems.🙂

**Comment by lachlan-00** [comment]
just testing on mysql again today. will fix tonight

**Comment by douglasriper** [comment]
Thank you, we'll see the next update (❛ᴗ❛)

**Comment by lachlan-00** [comment]
@douglasriper the fixes are in #1957.
Specifically the most recent commit
https://github.com/ampache/ampache/pull/1957/commits/bd7e45448b05fb8a7c5f82c185b5fae310189689

this pull will be merging soon as it's feature complete but you should download the new album/stats class to test them out. i've tested on mysql 8 and mariadb.

**Comment by douglasriper** [comment]
Hello, after updating the dev version the problem seems fixed for good. :)

_Sent from my Redmi  S2 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_
